framework/src/executioners/Transient.C

```diff
@@ -476,7 +476,7 @@ Transient::endStep(Real input_time)
 
   _last_solve_converged = lastSolveConverged();
 
-  if (_last_solve_converged && !_xfem_repeat_step)
+  if (_last_solve_converged && !_xfem_repeat_step && _time_stepper->enableOutput())
```

Comment:
maybe just move this `_time_stepper->enableOutput()` before `_problem.outputStep()`?